--- dwm.c.backup	2013-03-15 22:11:35.719963000 +0530
+++ dwm.c	2013-03-15 22:17:59.651217107 +0530
@@ -219,8 +219,10 @@ static void maprequest(XEvent *e);
 static void monocle(Monitor *m);
 static void motionnotify(XEvent *e);
 static void movemouse(const Arg *arg);
+static void nextlayout(const Arg *arg);
 static Client *nexttiled(Client *c);
 static void pop(Client *);
+static void prevlayout(const Arg *arg);
 static void propertynotify(XEvent *e);
 static void quit(const Arg *arg);
 static Monitor *recttomon(int x, int y, int w, int h);
@@ -1445,6 +1447,26 @@ propertynotify(XEvent *e) {
 }
 
 void
+nextlayout(const Arg *arg) {
+    Layout *l;
+    for (l=(Layout *)layouts;l != selmon->lt[selmon->sellt];l++);
+    if (l->symbol && (l + 1)->symbol)
+        setlayout(&((Arg) { .v = (l + 1) }));
+    else
+        setlayout(&((Arg) { .v = layouts }));
+}
+
+void
+prevlayout(const Arg *arg) {
+    Layout *l;
+    for (l=(Layout *)layouts;l != selmon->lt[selmon->sellt];l++);
+    if (l != layouts && (l - 1)->symbol)
+        setlayout(&((Arg) { .v = (l - 1) }));
+    else
+        setlayout(&((Arg) { .v = &layouts[LENGTH(layouts) - 2] }));
+}
+
+void
 quit(const Arg *arg) {
 	running = False;
 }
